### PR TITLE
NazarYakh_Xiaomi_search_test

### PR DESCRIPTION
**Pull request by NazarYakh**

### CODE REVIEW

File: src/main/java/com/ss/ita/rozetka/ui/pages/ProductTypePage.java [hunk 1]
```diff
@@ -17,4 +16,8 @@ public ProductTypePage chooseSubCategory(ProductCategoryAndSubCategory subCatego
         $x(format("//a[contains(@href,'%s')]", subCategory.getName())).click();
         return this;
     }
+
+    public String getProductTitle(int productNumber) {
+        return $$("span.goods-tile__title").get(productNumber).getText();
```
Comment (toleksyn) [inline src/main/java/com/ss/ita/rozetka/ui/pages/ProductTypePage.java:21]: use direct locators, not $$x().get(number)

File: src/test/java/com/ss/ita/rozetka/ui/XiaomiSearchTest.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import com.ss.ita.rozetka.ui.runner.TestRunner;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class XiaomiSearchTest extends TestRunner {
+    @Test
+    public void verifyXiaomiWillBeInSearchResult() {
+        new HomePage().open().header.doSearch("Xiaomi");
+
+        Assert.assertTrue(new ProductTypePage().getProductTitle(1).contains("Xiaomi"));
```
Comment (toleksyn) [inline src/test/java/com/ss/ita/rozetka/ui/XiaomiSearchTest.java:14]: where's the PO chain?

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/test/java/com/ss/ita/rozetka/ui/XiaomiSearchTest.java (hunk 2 above)
Comment (toleksyn) [reply src/test/java/com/ss/ita/rozetka/ui/XiaomiSearchTest.java:14]: remove duplication

File: src/main/java/com/ss/ita/rozetka/ui/pages/ProductTypePage.java [hunk 3]
```diff
@@ -17,4 +17,8 @@ public ProductTypePage chooseSubCategory(ProductCategoryAndSubCategory subCatego
         $x(format("//a[contains(@href,'%s')]", subCategory.getName())).click();
         return this;
     }
+
+    public String getFirstProductTitle() {
```
Comment (toleksyn) [inline src/main/java/com/ss/ita/rozetka/ui/pages/ProductTypePage.java:21]: NO hard coding

File: src/test/java/com/ss/ita/rozetka/ui/XiaomiSearchTest.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+import com.ss.ita.rozetka.ui.pages.HomePage;
+import com.ss.ita.rozetka.ui.pages.ProductTypePage;
+import com.ss.ita.rozetka.ui.runner.TestRunner;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class XiaomiSearchTest extends TestRunner {
+
+    String searchProduct = "Xiaomi";
```
Comment (toleksyn) [inline src/test/java/com/ss/ita/rozetka/ui/XiaomiSearchTest.java:11]: ???

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/test/java/com/ss/ita/rozetka/ui/XiaomiSearchTest.java (hunk 4 above)
Comment (toleksyn) [reply src/test/java/com/ss/ita/rozetka/ui/XiaomiSearchTest.java:11]: inline

File: src/test/java/com/ss/ita/rozetka/ui/XiaomiSearchTest.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
+package com.ss.ita.rozetka.ui;
+
+import com.ss.ita.rozetka.ui.pages.HomePage;
+import com.ss.ita.rozetka.ui.pages.ProductTypePage;
+import com.ss.ita.rozetka.ui.runner.TestRunner;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class XiaomiSearchTest extends TestRunner {
```
Comment (toleksyn) [inline src/test/java/com/ss/ita/rozetka/ui/XiaomiSearchTest.java:9]: rename

File: src/test/java/com/ss/ita/rozetka/ui/XiaomiSearchTest.java [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    @Test
+    public void verifyXiaomiWillBeInSearchResult() {
+        String searchProduct = "Xiaomi";
+
+        ProductTypePage productTypePage = new HomePage()
+                .open()
+                .getHeader()
+                .doSearch(searchProduct);
+
+        Assert.assertTrue(productTypePage.getProductTitle(1).contains(searchProduct));
```
Comment (toleksyn) [inline src/test/java/com/ss/ita/rozetka/ui/XiaomiSearchTest.java:20]: assertj

File: src/main/java/com/ss/ita/rozetka/ui/pages/ProductTypePage.java [hunk 7]
```diff
@@ -17,4 +17,8 @@ public ProductTypePage chooseSubCategory(ProductCategoryAndSubCategory subCatego
         $x(format("//a[contains(@href,'%s')]", subCategory.getName())).click();
         return this;
     }
+
+    public String getProductTitle(int productNumber) {
```
Comment (toleksyn) [inline src/main/java/com/ss/ita/rozetka/ui/pages/ProductTypePage.java:21]: step?

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/test/java/com/ss/ita/rozetka/ui/SearchFunctionalityTests.java [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+    public void verifyXiaomiWillBeInSearchResult() {
+        String searchProduct = "Xiaomi";
+
+        ProductTypePage productTypePage = new HomePage()
+                .open()
+                .getHeader()
+                .doSearch(searchProduct);
+
+        assertThat(productTypePage.getProductTitle(1))
+                .as("Product should have Xiaomi in title").contains(searchProduct);
```
Comment (toleksyn) [inline src/test/java/com/ss/ita/rozetka/ui/SearchFunctionalityTests.java:23]: format things properly

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/test/java/com/ss/ita/rozetka/ui/SearchFunctionalityTests.java [hunk 9]
```diff
@@ -0,0 +1,25 @@
+package com.ss.ita.rozetka.ui;
+
+import com.ss.ita.rozetka.ui.pages.HomePage;
+import com.ss.ita.rozetka.ui.pages.ProductTypePage;
+import com.ss.ita.rozetka.ui.TestUtils.TestRunner;
+import org.testng.annotations.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class SearchFunctionalityTests extends TestRunner {
```
Comment (toleksyn) [inline src/test/java/com/ss/ita/rozetka/ui/SearchFunctionalityTests.java:10]: Test, singular

**Review by toleksyn — CHANGES_REQUESTED** [review]

**Review by dzhyv — APPROVED** [review]

**Pull request merged**